hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -112,7 +114,35 @@ withClusterTempDir name =
   withTempDir ("hydra-cluster-e2e-" <> name)
 
 spec :: Spec
-spec = around (showLogsOnFailure "EndToEndSpec") $
+spec = around (showLogsOnFailure "EndToEndSpec") $ do
+  it "End-to-end offline mode" $ \tracer -> do
+    withTempDir ("offline-mode-e2e") $ \tmpDir -> do
+      let networkId = Testnet (NetworkMagic 42) -- from defaultChainConfig
+      let startingState =
+            [ (Alice, lovelaceToValue 100_000_000)
+            , (Bob, lovelaceToValue 100_000_000)
+            ]
+      (aliceCardanoVk, aliceCardanoSk) <- keysFor Alice
+      (bobCardanoVk, _) <- keysFor Bob
+      offlineConfig <- offlineConfigFor startingState tmpDir networkId
+
+      initialUtxo <- Aeson.throwDecodeStrict @UTxO.UTxO =<< readFileBS (initialUTxOFile offlineConfig)
+      let Just (aliceSeedTxIn, aliceSeedTxOut) = UTxO.find (\(TxOut addr _ _ _) -> addr == mkVkAddress networkId aliceCardanoVk) initialUtxo
```

Comment:
Could: use scenario values instead of loading/inferring them again here.

For example given my comment above, this could be simplified by just keeping `aliceInitialUTxO <- generate $ genUTxOFor aliceCardanoVk` around and there is also `UTxO.min` which will give you one (no filtering needed if we generate only for alice).